services/slam/fake/slam.go

```diff
@@ -115,29 +155,23 @@ func (slamSvc *SLAM) GetMap(ctx context.Context, name, mimeType string, cp *refe
 
 // Position returns a PoseInFrame of the robot's current location according to SLAM.
 func (slamSvc *SLAM) Position(ctx context.Context, name string, extra map[string]interface{}) (*referenceframe.PoseInFrame, error) {
-	path := filepath.Clean(artifact.MustPath(fmt.Sprintf(positionTemplate, slamSvc.dataCount)))
+	path := filepath.Clean(artifact.MustPath(fmt.Sprintf(positionTemplate, slamSvc.count())))
 	slamSvc.logger.Debug("Reading " + path)
 	data, err := os.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
+	position := position{}
+	err = json.Unmarshal(data, &position)
 
-	substrings := strings.Split(string(data), " | ")
-
-	rawXYZ, err := extract(strings.Split(substrings[0], " "))
-	if err != nil {
-		return nil, err
-	}
-	xyz := r3.Vector{X: rawXYZ[0], Y: rawXYZ[1], Z: rawXYZ[2]}
-
-	rawAA, err := extract(strings.Split(substrings[1], " "))
 	if err != nil {
 		return nil, err
 	}
+	xyz := r3.Vector{X: position.Pose.Pose.X, Y: position.Pose.Pose.Y, Z: position.Pose.Pose.Z}
 
-	axisAngle := spatialmath.NewR4AA().AxisAngles()
-	axisAngle.RX, axisAngle.RY, axisAngle.RZ, axisAngle.Theta = rawAA[0], rawAA[1], rawAA[2], rawAA[3]
-	pose := spatialmath.NewPose(xyz, axisAngle)
+	quat := position.Extra.Quat
```

Comment:
This changed as the new dataset was collected via grpcurl which persists the data as quats as the C++ outputs quats.